nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-processor.js

```diff
@@ -547,30 +557,64 @@ nf.Processor = (function () {
                     details.remove();
                 }
             }
-        });
-        
-        // ---------------
-        // processor color
-        // ---------------
-
-        // update the processor color
-        updated.select('text.processor-icon')
-            .style('fill', function (d) {
-                
-                // get the default color
-                var color = nf.Processor.defaultColor();
-                
-                if (!d.permissions.canRead) {
-                    return color;
-                }
 
-                // use the specified color if appropriate
-                if (nf.Common.isDefinedAndNotNull(d.component.style['background-color'])) {
-                    color = d.component.style['background-color'];
-                }
+            // ---------------
+            // processor color
+            // ---------------
 
-                return color;
-            });
+            // update the processor color
+            processor.select('text.processor-icon')
+                .style('fill', function (d) {
+
+                    // get the default color
+                    var color = nf.Processor.defaultColor();
+
+                    if (!d.permissions.canRead) {
+                        //update the processor icon container
+                        processor.select('rect.processor-icon-container').classed('unauthorized', true);
```

Comment:
I don't think we should be updating/modifying these elements within the callback for a different element. Because we're already iterating through each matched processor, `d` is already available as `processorData`. Each of these` processor.select` should be brought outside of this callback.

FYI - we're using `.each` in this case because of how the tooltips are implemented. They didn't directly map into the traditional mapping process.